java/vector/src/main/java/org/apache/arrow/vector/complex/impl/UnionLargeListReader.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.vector.complex.impl;
+
+import static org.apache.arrow.memory.util.LargeMemoryUtil.checkedCastToInt;
+
+import org.apache.arrow.vector.ValueVector;
+import org.apache.arrow.vector.complex.LargeListVector;
+import org.apache.arrow.vector.complex.reader.FieldReader;
+import org.apache.arrow.vector.complex.writer.FieldWriter;
+import org.apache.arrow.vector.holders.UnionHolder;
+import org.apache.arrow.vector.types.Types.MinorType;
+import org.apache.arrow.vector.types.pojo.Field;
+
+/**
+ * {@link FieldReader} for list of union types.
+ */
+public class UnionLargeListReader extends AbstractFieldReader {
+
+  private LargeListVector vector;
+  private ValueVector data;
+  private long index;
+  private static final int OFFSET_WIDTH = 8;
+
+  public UnionLargeListReader(LargeListVector vector) {
+    this.vector = vector;
+    this.data = vector.getDataVector();
+  }
+
+  @Override
+  public Field getField() {
+    return vector.getField();
+  }
+
+  @Override
+  public boolean isSet() {
+    return !vector.isNull(checkedCastToInt(index));
+  }
+
+  private long currentOffset;
+  private long maxOffset;
+
+  @Override
+  public void setPosition(int index) {
+    setIndex(index);
+  }
+
+  /**
+   * set current index.
+   *
+   * This is different from setPosition in that it supports int64
+   */
+  public void setIndex(long index) {
```

Comment:
why is this needed? the vector only supports integer max_value number of elements right?

Reply:
you are correct. Removed